docs/docs/learn/programming/mcp.md

```diff
@@ -0,0 +1,187 @@
+
+The most common way to use MCP is with a local server process communicating via stdio:
+
+```python
+import asyncio
+import dspy
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+
+async def main():
+    # Configure the stdio server
+    server_params = StdioServerParameters(
+        command="python",                    # Command to run
+        args=["path/to/your/mcp_server.py"], # Server script path
+        env=None,                            # Optional environment variables
+    )
+
+    # Connect to the server
+    async with stdio_client(server_params) as (read, write):
+        async with ClientSession(read, write) as session:
+            # Initialize the session
+            await session.initialize()
+
+            # List available tools
+            response = await session.list_tools()
+
+            # Convert MCP tools to DSPy tools
+            dspy_tools = [
+                dspy.Tool.from_mcp_tool(session, tool)
+                for tool in response.tools
+            ]
+
+            # Create a ReAct agent with the tools
+            class QuestionAnswer(dspy.Signature):
+                """Answer questions using available tools."""
+                question: str = dspy.InputField()
+                answer: str = dspy.OutputField()
+
+            react_agent = dspy.ReAct(
+                signature=QuestionAnswer,
+                tools=dspy_tools,
+                max_iters=5
+            )
+
+            # Use the agent
+            result = await react_agent.acall(
+                question="What is 25 + 17?"
+            )
+            print(result.answer)
+
+# Run the async function
+asyncio.run(main())
+```
+
+### 2. HTTP Server (Remote)
+
+For remote MCP servers over HTTP, use the streamable HTTP transport:
+
+```python
+import asyncio
+import dspy
+from mcp import ClientSession
+from mcp.client.streamable_http import streamablehttp_client
+
+async def main():
+    # Connect to HTTP MCP server
+    async with streamablehttp_client("http://localhost:8000/mcp") as (read, write):
+        async with ClientSession(read, write) as session:
+            # Initialize the session
+            await session.initialize()
+
+            # List and convert tools
+            response = await session.list_tools()
+            dspy_tools = [
+                dspy.Tool.from_mcp_tool(session, tool)
+                for tool in response.tools
+            ]
+
+            # Create and use ReAct agent
+            class TaskSignature(dspy.Signature):
+                task: str = dspy.InputField()
+                result: str = dspy.OutputField()
+
+            react_agent = dspy.ReAct(
+                signature=TaskSignature,
+                tools=dspy_tools,
+                max_iters=5
+            )
+
+            result = await react_agent.acall(task="Check the weather in Tokyo")
+            print(result.result)
+
+asyncio.run(main())
+```
+
+## Creating an MCP Server
```

Comment:
Do we need this section to be a standalone section? maybe we can move the content under "## Tool Conversion", and say something like "Let's build a simple MCP server with tools and surface the MCP tools to DSPy"